mlrun/projects/project.py

```diff
@@ -1211,8 +1211,14 @@ def set_workflow(
         :param ttl:           pipeline ttl in secs (after that the pods will be removed)
         :param args:          argument values (key=value, ..)
         """
-        if not workflow_path:
-            raise ValueError("valid workflow_path must be specified")
+
+        if mlrun.utils.helpers.is_file_path_invalid(
+            self.spec.get_code_path(), workflow_path
+        ):
+            raise ValueError(
+                f"Invalid 'workflow_path': '{workflow_path}'. Please provide a valid URL/path to a file."
+            )
```

Comment:
move the raise exception to the utils method and then you only need to call it here without the if
```suggestion
        mlrun.utils.helpers.validate_file_path(self.spec.get_code_path(), workflow_path)
```